backend/src/main/java/fk/prof/backend/http/BackendHttpVerticle.java

```diff
@@ -117,13 +136,55 @@ private void handlePostProfile(RoutingContext context) {
         });
   }
 
+  private void handlePostPoll(RoutingContext context) {
+    try {
+      Recorder.PollReq pollReq = ProtoUtil.buildProtoFromBuffer(Recorder.PollReq.parser(), context.getBody());
+      Recorder.ProcessGroup processGroup = RecorderProtoUtil.mapRecorderInfoToProcessGroup(pollReq.getRecorderInfo());
+      ProcessGroupContextForPolling processGroupContextForPolling = this.processGroupDiscoveryContext.getProcessGroupContextForPolling(processGroup);
+      if(processGroupContextForPolling == null) {
+        throw new IllegalArgumentException("Process group " + RecorderProtoUtil.processGroupCompactRepr(processGroup) + " not associated with the backend");
```

Comment:
So I am following a simple convention here. All code in request handlers is always wrapped in try catch where exceptions are mapped to HttpFailure (extends runtime exception). The response generation behaviour is defined by the http failure object.
Convention is that all illegalargumentexceptions are mapped to 400 error code, illegalstateexceptions are mapped to 500, and for custom exceptions like backendassociationexception, aggregationfailure, behavior is determined by their `isServerFailure` property